CMakeLists.txt

```diff
@@ -387,18 +387,63 @@ endmacro()
 
 if(UNIX)
   include(CheckCSourceCompiles)
+
+  # don't allow implicit function declarations
+  set(CMAKE_REQUIRED_FLAGS "-std=c99")
```

Comment:
This causes the `HAVE_CLOCK_GETTIME` check to fail on my system (with various versions of GCC).

Using `-std=gnu99` instead or adding `-D_POSIX_C_SOURCE=199309L` fixes it.

Reply:
There should be a ``#define _DEFAULT_SOURCE`` in the HAVE_CLOCK_GETTIME test with commit ff42b03 - according to gc manual for _DEFAULT_SOURCE, "the effect is to enable features from the 2008 edition of POSIX". Weird thing is, i can't see that commit (and a bunch of others) in this PR.

Reply:
`_DEFAULT_SOURCE` is there, but it was only introduced in glibc 2.19, and my glibc is a couple of revs older than that (using RHEL 7.3).

Reply:
@jrprice OK thanks, i forgot that _DEFAULT_SOURCE is quite recent.

Reply:
Setting `_POSIX_C_SOURCE` to 200809L should work too.